src/ex2/pokemonClient.js

```
@@ -0,0 +1,43 @@
+class PokemonClient {
+    constructor() {
+        this.API_BASE = `https://pokeapi.co/api/v2/`;
+    }
+
+
+    async catchPokemons(numbers) { // numbers -> array of numbers
+        const API = this.API_BASE + `pokemon/`;
+        const promises = [];
+        let res = [];
+        numbers.map((num) => {
+            promises.push(fetch(API + num + '/'));
+        })
+        try {
+            const response = await Promise.all(promises);
+            for (const item of response) {//resolve each promise
+                res.push(await item.json());
+            }
+            res.forEach((el, index) => res[index] = el.name);//extract each element to pokemon name
+            console.log('fetched pokemons: ', res);
+            return res;//one or more names
+        }
+        catch (e) {//Promise.all failed
+            console.log('Catch Error: ', e);
+            return false;
```

i think its better to throw the exception, when returning false its sometime confusing and if you'll want to differentiate later on between different errors it will be harder (like between no internet, and no such pokemon for example)